iroh-net/src/magic_endpoint.rs

```diff
@@ -411,6 +411,14 @@ impl MagicEndpoint {
 
         let NodeAddr { node_id, info } = node_addr;
 
+        // Connecting to ourselves is not supported.
+        if node_id == self.node_id() {
```

Comment:
Should you not do this before calling `.add_node_addr()`?

Reply:
Should we fail for `add_node_addr` with node_id == self.node_id too?

Reply:
yes, i think that would also be good.  in which case you could just rely on that to fail here.

Reply:
Added this. We can't only check there, because add_node_addr is only called if the NodeAddr has non-empty AddrInfo. But we also don't want to start discovery for ourself is AddrInfo is empty, so two checks for now.